src/commands.rs

```diff
@@ -542,39 +506,21 @@ fn handle_compile_response<T>(mut creator: T,
     where T : CommandCreatorSync,
 {
     match response {
-        CompileResponse::CompileStarted(_) => {
+        CompileResponse::CompileStarted => {
             debug!("Server sent CompileStarted");
             // Wait for CompileFinished.
-            match conn.read_one_response() {
-                Ok(mut res) => {
-                    if res.has_compile_finished() {
-                        trace!("Server sent CompileFinished");
-                        return handle_compile_finished(res.take_compile_finished(),
-                                                       stdout, stderr)
-                    } else {
-                        bail!("unexpected response from server")
-                    }
-                }
-
-                // Currently the shutdown behavior of the remote sccache server
```

Comment:
I don't see that you replaced this behavior anywhere (compile locally on EOF from server). Presumably you do want to keep this in some form?